webapp/graphite/render/views.py

```diff
@@ -461,6 +463,29 @@ def parseOptions(request):
   return (graphOptions, requestOptions)
 
 
+# extract headers which get set by Grafana when issuing queries, to help identifying where a query came from.
+# user-defined headers from settings.INPUT_VALIDATION_SOURCE_ID_HEADERS also get extracted and mixed
+# with the standard Grafana headers.
+def extractSourceIdHeaders(request):
+    headers = {}
+
+    if 'HTTP_X_GRAFANA_ORG_ID' in request.META:
+        headers['grafana-org-id'] = request.META['HTTP_X_GRAFANA_ORG_ID']
+
+    if 'HTTP_X_DASHBOARD_ID' in request.META:
+        headers['dashboard-id'] = request.META['HTTP_X_DASHBOARD_ID']
+
+    if 'HTTP_X_PANEL_ID' in request.META:
+        headers['panel-id'] = request.META['HTTP_X_PANEL_ID']
```

Comment:
You could handle the grafana headers at the same time as the custom headers, maybe like ...

```python
source_headers = {
    'grafana-org-id': 'X-Grafana-Org-ID',
    'dashboard-id':   'X-Dashboard-ID',
    'panel-id':       'X-Panel-ID',
}
source_headers.update(settings.INPUT_VALIDATION_SOURCE_ID_HEADERS)

for hdr_name, log_name in source_headers.items():
   value = request.META.get('HTTP_' + hdr_name.upper().replace('-', '_'))
    if value:
        headers[log_name] = value
```

Reply:
Good call

Reply:
yep, good idea, i'll do that

Reply:
updated: https://github.com/graphite-project/graphite-web/pull/2564/commits/a735ba21c7ba210df3274e10d7c81e5df3bba8a7

Reply:
looks good to me, thanks :)

Reply:
thanks to you, i'll go ahead and merge then